tst/EnergyPlus/unit/FuelCellElectricGenerator.unit.cc

```diff
@@ -55,132 +55,970 @@
+
+    EXPECT_EQ("NODE 10", exhaustHX.WaterInNodeName);
+    EXPECT_GT(exhaustHX.WaterInNode, 0);
+
+    EXPECT_EQ("NODE 11", exhaustHX.WaterOutNodeName);
+    EXPECT_GT(exhaustHX.WaterOutNode, 0);
+
+    EXPECT_EQ(0.0004, exhaustHX.WaterVolumeFlowMax);
+
+    EXPECT_EQ("GENERATOR FUEL CELL EXHAUST GAS TO WATER HEAT EXCHANGER 1 EXHAUST OUTLET AIR NODE", exhaustHX.ExhaustOutNodeName);
+    EXPECT_GT(exhaustHX.ExhaustOutNode, 0);
+
+    EXPECT_EQ(DataGenerators::ExhaustGasHX::Condensing, thisFC->ExhaustHX.HXmodelMode);
+
+    EXPECT_EQ(83.1, exhaustHX.hxs0);
+    EXPECT_EQ(4798.0, exhaustHX.hxs1);
+    EXPECT_EQ(-138000.0, exhaustHX.hxs2);
+    EXPECT_EQ(-353800.0, exhaustHX.hxs3);
+    EXPECT_EQ(515000000.0, exhaustHX.hxs4);
+
+    EXPECT_EQ(0.0, exhaustHX.HXEffect);
+    EXPECT_EQ(0.0, exhaustHX.NdotGasRef);
+    EXPECT_EQ(0.0, exhaustHX.nCoeff);
+    EXPECT_EQ(0.0, exhaustHX.AreaGas);
+    EXPECT_EQ(0.0, exhaustHX.h0Water);
+    EXPECT_EQ(0.0, exhaustHX.NdotWaterRef);
+    EXPECT_EQ(0.0, exhaustHX.mCoeff);
+    EXPECT_EQ(0.0, exhaustHX.AreaWater);
+    EXPECT_EQ(0.0, exhaustHX.Fadjust);
+    EXPECT_EQ(0.0031, exhaustHX.l1Coeff);
+    EXPECT_EQ(1.0, exhaustHX.l2Coeff);
+    EXPECT_EQ(35.0, exhaustHX.CondensationThresholdTemp);
+
+
+    // Electrical Storage
+    EXPECT_EQ("GENERATOR FUEL CELL ELECTRICAL STORAGE 1", thisFC->NameElecStorage);
+    auto elecStorage = thisFC->ElecStorage;
+    EXPECT_EQ(elecStorage.Name, thisFC->NameElecStorage);
+    EXPECT_EQ(DataGenerators::ElectricalStorage::SimpleEffConstraints, elecStorage.StorageModelMode);
+    EXPECT_EQ(1.0, elecStorage.EnergeticEfficCharge);
+    EXPECT_EQ(1.0, elecStorage.EnergeticEfficDischarge);
+    EXPECT_EQ(0.0, elecStorage.NominalEnergyCapacity);
+    EXPECT_EQ(0.0, elecStorage.MaxPowerDraw);
+    EXPECT_EQ(0.0, elecStorage.MaxPowerStore);
+    EXPECT_EQ(0.0, elecStorage.StartingEnergyStored);
+
+
+
+    // Inverter
+    EXPECT_EQ("GENERATOR FUEL CELL INVERTER 1", thisFC->NameInverter);
+    auto inverter = thisFC->Inverter;
+    EXPECT_EQ(DataGenerators::InverterEfficiencyMode::Constant, inverter.EffMode);
+    EXPECT_EQ(1.0, inverter.ConstEff);
+    ASSERT_GT(inverter.EffQuadraticCurveID, 0);
+    EXPECT_EQ("EFFICIENCY FUNCTION OF DC POWER CURVE", state->dataCurveManager->PerfCurve(inverter.EffQuadraticCurveID).Name);
+
+    // StackCooler: not included
+
+    // other checks
+    EXPECT_EQ(1, thisFC->CWLoopNum);
+
+    auto report = thisFC->Report;
+    EXPECT_EQ(exhaustHX.qHX, report.qHX);
+    EXPECT_EQ(report.ACPowerGen, generatorController->electProdRate);
+    EXPECT_EQ(report.ACEnergyGen, generatorController->electricityProd);
+
+    EXPECT_TRUE(generatorController->electProdRate > 1.15 * generatorController->thermProdRate) << "Power to Heat Ratio appears too low";
+    EXPECT_DOUBLE_EQ(exhaustHX.qHX, generatorController->thermProdRate);
+    EXPECT_DOUBLE_EQ(generatorController->thermProdRate * 15 * 60, generatorController->thermalProd);
 }
```

Comment:
Thank you so much for finishing this huge unit test and catching two more defects about this while at it.  We appreciate it.